youtube_dl/extractor/rockstargames.py

```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    qualities,
+    parse_iso8601
+)
+
+
+class RockstarGamesIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?rockstargames\.com/videos/video/(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'https://www.rockstargames.com/videos/video/11544/',
+        'md5': '03b5caa6e357a4bd50e3143fc03e5733',
+        'info_dict': {
+            'id': '11544',
+            'ext': 'mp4',
+            'title': 'Further Adventures in Finance and Felony Trailer',
+            'thumbnail': 're:^https?://.*\.jpg$',
+            'description': 'md5:6d31f55f30cb101b5476c4a379e324a3',
+            'upload_date': '20160602',
+            'timestamp': 1464876000
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        json_data = self._download_json(
+            'https://www.rockstargames.com/videoplayer/videos/get-video.json?id=%s&locale=en_us' % video_id,
+            video_id
+        )['video']
+
+        formats = []
+
+        for video in json_data['files_processed']['video/mp4']:
+            if not video.get('src'):
+                continue
+            height = video.get('resolution', '').replace('p', '')
+            
+            formats.append({
+                'url': self._proto_relative_url(video['src']),
+                'quality': int(height) if height.isdigit() else -1,
```

Comment:
Boom, done

On Mon, Jun 13, 2016 at 7:41 PM, Sergey M. <notifications@github.com> wrote:

> In youtube_dl/extractor/rockstargames.py
> <https://github.com/rg3/youtube-dl/pull/9737#discussion_r66834783>:
>
> > +        video_id = self._match_id(url)
> > +        json_data = self._download_json(
> > +            'https://www.rockstargames.com/videoplayer/videos/get-video.json?id=%s&locale=en_us' % video_id,
> > +            video_id
> > +        )['video']
> > +
> > +        formats = []
> > +
> > +        for video in json_data['files_processed']['video/mp4']:
> > +            if not video.get('src'):
> > +                continue
> > +            height = video.get('resolution', '').replace('p', '')
> > +
> > +            formats.append({
> > +                'url': self._proto_relative_url(video['src']),
> > +                'quality': int(height) if height.isdigit() else -1,
>
> and put as height field.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/rg3/youtube-dl/pull/9737/files/66c8294d75388f68b46c7d7bf4a2b1b9b62137a3#r66834783>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC-3KO7LDL9cbCetSJZJ0jsQ95efvUGNks5qLZZbgaJpZM4Ix4UC>
> .
>